### DISCUSSION TITLE
how to stop my clients to disappear when restarting the server ?

### DISCUSSION THREAD

**Discussion by Amanb1145** [ory/hydra#3624] [Q&A]
Hello everyone,

Whenever I restart or rebuild my server, the previously created clients disappear. Does anyone have any ideas on how to prevent this from happening? Thanks!

**Comment by vinckr** [comment]
Hello @Amanb1145 

Hard to say what is exactly going on without more information, but you are probably using the in-memory option, so `dns` is set to `memory`. 

To persist data, please attach a [database](https://www.ory.sh/docs/self-hosted/deployment) like Postgres. 

If you are new to self-hosting services, I would recommend to check out [Ory Network](https://console.ory.sh/projects/current/dashboard) - its the same tech, but convenient to use. Self-hosting in general has a steep learning curve.

**Comment by Amanb1145** [comment]
Hey @vinckr Thanks for responding and I will definitely checkout Ory Network
However I noticed something-
I am starting the server using-
`docker-compose -f quickstart.yml \
    -f quickstart-postgres.yml \
    up --build`
  but in `quickstart.yml` sqlite db is defined is the issue ? is this the issue ?
  `
version: "3.7"
services:
  hydra:
    image: oryd/hydra:v2.2.0-rc.2
    ports:
      - "4444:4444" # Public port
      - "4445:4445" # Admin port
      - "5555:5555" # Port for hydra token user
    command: serve -c /etc/config/hydra/hydra.yml all --dev
    volumes:
      - type: volume
        source: hydra-sqlite
        target: /var/lib/sqlite
        read_only: false
      - type: bind
        source: ./contrib/quickstart/5-min
        target: /etc/config/hydra
    environment:
      - DSN=sqlite:///var/lib/sqlite/db.sqlite?_fk=true
    restart: unless-stopped
    depends_on:
      - hydra-migrate
    networks:
      - intranet
  hydra-migrate:
    image: oryd/hydra:v2.2.0-rc.2
    environment:
      - DSN=sqlite:///var/lib/sqlite/db.sqlite?_fk=true
    command: migrate -c /etc/config/hydra/hydra.yml sql -e --yes
    volumes:
      - type: volume
        source: hydra-sqlite
        target: /var/lib/sqlite
        read_only: false
      - type: bind
        source: ./contrib/quickstart/5-min
        target: /etc/config/hydra
    restart: on-failure
    networks:
      - intranet
  consent:
    environment:
      - HYDRA_ADMIN_URL=http://hydra:4445
    image: oryd/hydra-login-consent-node:v2.2.0-rc.2
    ports:
      - "3000:3000"
    restart: unless-stopped
    networks:
      - intranet
networks:
  intranet:
volumes:
  hydra-sqlite:
`

**Reply by vinckr** [reply]
I see. It should persist with sqlite. 
Last time I tried out the quickstart it persisted clients etc. for me 🤔 
Did you just run the quickstart or change anything?

**Reply by Amanb1145** [reply]
No I have not changes anything.
whenever I try it on local for test purpose clients always persist for me but when I host it on a server then clients never persist

**Reply by vinckr** [reply]
How do you host it?

**Reply by Amanb1145** [reply]
I followed 5-min quickstart guide and it works perfectly if container is not stopped. I have hosted it on a azure vm instance with ubuntu 20.04.
